calculation/spatial_hazard/spatial_hazard/correlate_ims.py

```diff
@@ -5,65 +5,64 @@
 from scipy.linalg import cholesky
 
 import gmhazard_calc as gc
-from spatial_hazard import models
+import sha_calc
 from IM_calculation.source_site_dist.src_site_dist import calc_rrup_rjb
 
 
-def load_stations_data(imdb_ffps: Sequence[str], stations: Sequence[str], im: gc.im.IM):
-    """Loads the IM data for the specified stations from the IMDBs"""
-    result_dict = {station: [] for station in stations}
-    for imdb_ffp in imdb_ffps:
-        print("Getting {}".format(imdb_ffp))
-        with gc.dbs.IMDB.get_imdb(imdb_ffp, writeable=False) as imdb:
-            imdb_stations = imdb.sites().index
-            for station in stations:
-                if station not in imdb_stations:
-                    print("Station {} has no entries here".format(station))
-                cur_data = imdb.im_data(station, im, inter_intra=True)
-                if cur_data is not None:
-                    result_dict[station].append(cur_data)
-
-    result_dict = {key: pd.concat(value) for key, value in result_dict.items()}
-    return result_dict
-
-
 def load_stations_fault_data(
     imdb_ffps: Sequence[str], stations: Sequence[str], im: gc.im.IM, fault: str
 ):
-    """Loads the IM data for the specified stations and fault from the IMDBs"""
-    stations_data = load_stations_data(imdb_ffps, stations, im)
-
-    return pd.DataFrame(
-        {
-            cur_station: cur_df.loc[fault].to_dict()
-            for cur_station, cur_df in stations_data.items()
-        }
-    ).T
-
+    """
+    Loads the IM data for the specified stations and fault from the IMDB's
 
-def calculate_distance_matrix(
-    stations: Sequence[str], locations_df: pd.DataFrame
-) -> pd.DataFrame:
+    Parameters
+    ----------
+    imdb_ffps: Sequence[str]
+        List of imdb full file paths to load
+    stations: Sequence[str]
+        List of station names to grab data for from the imdbs
+    im: IM
+        IM Object to extract values from the imdb
+    fault: String
+        Fault name to extract data from the imdb
+    """
+    # Obtain the given IMDB with the correct fault information
```

Comment:
No `given`?
Also I would also make clear here, that the assumption we are making is that for a given fault there is only a single IMDB that will contain the required data. Due to the fact that spatial hazard does not support Ensembles, i.e. only works for a single GMM.

Reply:
@joelridden Thoughts on adding extra comment regarding assumptions made?